### PR TITLE
Detect pseudo/v3

### PR DESCRIPTION
**Pull request by victorjulien**
https://redmine.openinfosecfoundation.org/issues/7002 with a few minor cleanups.

Replace #11001:
- rebase to master
- add csum
- stream_size fixup

### CODE REVIEW

File: src/detect-stream_size.c [hunk 1]
```diff
@@ -170,7 +169,8 @@ void DetectStreamSizeFree(DetectEngineCtx *de_ctx, void *ptr)
 static void PrefilterPacketStreamsizeMatch(
         DetectEngineThreadCtx *det_ctx, Packet *p, const void *pectx)
 {
-    if (!(PacketIsTCP(p)) || PKT_IS_PSEUDOPKT(p))
+    DEBUG_VALIDATE_BUG_ON(PKT_IS_PSEUDOPKT(p));
```
Comment (victorjulien) [inline src/detect-stream_size.c:172]: seems wrong

Comment (catenacyber) [reply src/detect-stream_size.c:172]: Is this PR a draft then ?

File: src/detect-stream_size.c [hunk 2]
```diff
@@ -214,9 +212,9 @@ static bool PrefilterPacketStreamSizeCompare(PrefilterPacketHeaderValue v, void
 
 static int PrefilterSetupStreamSize(DetectEngineCtx *de_ctx, SigGroupHead *sgh)
 {
-    return PrefilterSetupPacketHeader(de_ctx, sgh, DETECT_STREAM_SIZE,
-            SIG_MASK_REQUIRE_FLOW | SIG_MASK_REQUIRE_REAL_PKT, PrefilterPacketStreamSizeSet,
-            PrefilterPacketStreamSizeCompare, PrefilterPacketStreamsizeMatch);
+    return PrefilterSetupPacketHeader(de_ctx, sgh, DETECT_STREAM_SIZE, SIG_MASK_REQUIRE_FLOW,
```
Comment (catenacyber) [inline src/detect-stream_size.c:215]: This last commit needs to be squashed, right ?

**Comment by catenacyber** [comment]
There seems to be that there is a big feature that prefilter uses `SIG_MASK_REQUIRE_FLOW` and other flags now (not only `SIG_MASK_REQUIRE_REAL_PKT`) right ?

**Comment by victorjulien** [comment]
replaced by #11274

**Pull request closed without merging**